src/Pay/Adapter/Stripe.php

```diff
@@ -0,0 +1,223 @@
+            $requestBody['exp_month'] = $expMonth;
+        }
+        if(!empty($expYear)) {
+            $requestBody['exp_year'] = $expYear;
+        }
+        return $this->execute('POST', $path, $requestBody);
+    }
+
+    /**
+     * Get a card
+     */
+    public function getCard(string $customerId, string $cardId): array {
+        $path = '/customers/' . $customerId . '/sources/' . $cardId;
+        return $this->execute('GET', $path);
+    }
+
+    /**
+     * Delete a credit card record
+     */
+    public function deleteCard(string $customerId, string $cardId) : bool {
+        $path = '/customers/' . $customerId . '/sources/' . $cardId;
+        $res =  $this->execute('DELETE', $path);
+        return $res['deleted'] ?? false;
+    }
+  
+    /**
+     * Add new customer in the gateway database
+     * returns the id of the newly created customer
+     * 
+     * @throws Exception
+     */
+    public function createCustomer(string $name, string $email, array $billingDetails = [], string $paymentMethod = null) : array {
+        $path = '/customers';
+        $requestBody = [
+            'name' => $name,
+            'email' => $email,
+        ];
+        if(!empty($paymentMethod)) {
+            $requestBody['payment_method'] = $paymentMethod;
+        }
+        if(!empty($billingDetails)) {
+            $requestBody['billing_details'] = $billingDetails;
+        }
+        $result = $this->execute('POST', $path, $requestBody);
+        return $result;
+    }
+
+    /**
+     * List customers
+     */
+    public function listCustomers(): array
+    {
+        return $this->execute('GET', '/customers');
+    }
+  
+    /**
+     * Get customer details by ID
+     */
+    public function getCustomer(string $customerId) : array {
+        $path = '/customers/' . $customerId;
+        $result = $this->execute('GET', $path, [], []);
+        return $result;
+    }
+  
+    /**
+     * Update customer details
+     */
+    public function updateCustomer(string $customerId, string $name, string $email,  array $billingDetails = [], string $paymentMethod = null) : array {
+        $path = '/customers/' . $customerId;
+        $requestBody = [
+            'name' => $name,
+            'email' => $email,
+        ];
+        if(!empty($paymentMethod)) {
+            $requestBody['payment_method'] = $paymentMethod;
+        }
+        if(!empty($billingDetails)) {
+            $requestBody['billing_details'] = $billingDetails;
+        }
+        return $this->execute('POST', $path, $requestBody);
+    }
+  
+    /**
+     * Delete customer by ID
+     */
+    public function deleteCustomer(string $customerId) : bool {
+        $path = '/customers/' . $customerId;
+        $result = $this->execute('DELETE', $path);
+        return $result['deleted'] ?? false;
+    }
+    
+    private function execute(string $method, string $path, array $requestBody = [], array $headers = ['content-type: application/x-www-form-urlencoded']) {
```

Comment:
Can we move the `execute` function to the base Adapter class ? 
Looks like all providers will need it

Reply:
might, but all providers will have different requirements, took this idea from the `utopia-php/logger` library

Reply:
Could be a good idea to extract the HTTP call kind of method to a higher level for life quality when writing new adapters. It could also be pretty generic and use protected attributes for recurring headers or anything similar.